### ISSUE TITLE
Normalization of input data

### ISSUE THREAD

**Issue by Yingyingzhang15** [fchollet/keras#2127]
Hi. I use a RNN to classify datas.My code is as follows:

timesteps = 50  
model = Sequential()
model.add(LSTM(50, input_shape=(timesteps, 3)))
model.add(Dropout(dropout))
model.add(Dense(3))
model.add(Activation('softmax'))
model.compile(loss='binary_crossentropy', optimizer='adam')

The sequence input_shape is (50,3), and I give each sequence a label.Actually, each (50,1)
 vector is a time sequence and they represents different aspects of my (50,3)  sequence.

Should I normalize all the 150 data to mean 0 and variance 1?
Or  should I normalize each (50,1) vector, or (1,3) vector? 

I think I should normalize each(1,3) data, but this will have a influence on each (50,1) time sequence.
Can you help me?Thank you!

**Comment by EderSantana** [comment]
It only makes sense to do a global normalization if the 3 series are in the same domain. Without knowing your data I would say it is safer to normalize each row independently.

**Comment by Yingyingzhang15** [comment]
@EderSantana My input data  (50,3)  is something like a three-channel (red,green,blue) image.  Each (50,1) vector represents a channel. Do I need to normalize all the 150 data?

**Comment by EderSantana** [comment]
it is safer to renormalize each (50,1) individually

**Comment by Yingyingzhang15** [comment]
@EderSantana Thank you very much!:)
